scripts/email/README.md

```diff
@@ -52,14 +52,49 @@ Then in comparison to other email services like Sendgrid, Resend stood out for i
     - `BCC` should be a list of emails to send to.
     - `SUBJECT` is the subject of the email.
     - `HTML` is the body of the email.
-3. **Test your template** by running `python scripts/email/base_template.py` with a simplified BCC list before sending to a large audience.
+3. **Test your template** by running `python3 scripts/email/execute_template.py` with a simplified BCC list before sending to a large audience.
 
 A couple notes:
 - You can refer to existing templates for best practices and to see how to e.g. have the `BCC` list be dynamically generated from our Firebase users.
 - **Important**: Ensure all HTML styling is inline as we unfortunately cannot use external CSS directives.
 
+## Fetching Users from Firebase
+
+We use Firebase to store user data and retrieve it for our email templates. The process is handled by the `firebase_users_loader.py` helper script under `scripts/email/helpers/`.
+
+1. The script connects to Firebase using a service account key, stored in the root directory of the project as `serviceAccountKey.json`.
+3. It retrieves user data from the `user-onboarding-data` collection.
+2. Then, it fetches all user names from the `user-name` collection.
+4. The data is processed and organized into a dictionary, with keys being tuples of (graduation_semester, graduation_year).
+5. Each user's data includes email, name, colleges, grad programs, majors, minors, and graduation information.
+
+The `USERS` variable in `firebase_users_loader.py` contains this processed data and is imported by individual email templates.
+
+### Using Firebase Data in Templates
+
+Email templates, such as `current_freshman.py`, import the `USERS` data and filter it based on specific criteria. For example:
+
+```python
+from scripts.email.templates.helpers.firebase_users_loader import USERS
+
+BCC = [
+    user["email"]
+    for users in USERS.values()
+    for user in users
+    if (
+        (user["graduation_year"] == "2028" and user["graduation_semester"] == "Spring")
+        or (user["graduation_year"] == "2027" and user["graduation_semester"] == "Fall")
+    )
+]
+```
+
+This code filters the `USERS` data to create a `BCC` list of email addresses for current freshmen (in FA24) based on their expected graduation year and semester.
+
+By using this approach, we can easily create targeted email lists for different groups of students without manually maintaining separate lists.
+
 ## Further Notes
 
+- You **must** run the script from the root directory of the project.
```

Comment:
Thank you for _such_ thorough documentation - future CP devs are going to appreciate this so much :)